### Bayes Theorem

The cigar example in the lab illustrates the application of Bayes' theorem with its calculation using the formula. Unfortunately, that calculation is complicated and can cause confusion and/or incorrect substitution of the involved
probability values. Fortunately, here is another approach that is much more intuitive and easier:

Assume some convenient value for the total of all items involved, then
construct a table of rows and columns with the individual cell frequencies
based on the known probabilities.

For example, let's assume that the adult population in Boone county is 100,000. Now we can use the given information to create a table.

*Number of males who smoke cigars:* 51% of adults are males; so there are 51,000 males. If 9.5% of them smoke, that makes 0.095 x 51,000 = 4845. Then, males who do not smoke are 51,000 - 4845 = 46,155. See the table where these values go.


*Number of females who smoke cigars:* 49% of the adults are females, that makes 49,000. 1.7% of them are smokers, so 0.017 x 49,000 = 833. The number of females who do not smoke is 49,000 - 833 = 48,167. Again look at the table below. 

In [15]:
cigar <- matrix(c(4845, 833, 46155, 48167), ncol=2)
colnames(cigar) <- c('smoker','nonsmoker')
rownames(cigar) <- c('male','female')
cigar.table <- as.table(cigar)

addmargins(cigar)

smoker nonsmoker    Sum
male     4845     46155  51000
female    833     48167  49000
Sum      5678     94322 100000

The above table involves simple arithmetic. Simply partition the
assumed population into the different cell categories by finding suitable percentages.

Now we can easily address the key question as follows: To find the probability of
getting a male subject, given that the subject smokes cigars, simply use the same
conditional probability described before. 

To find the probability of getting a
male given that the subject smokes, restrict the table to the column of cigar smokers, then
find the probability of getting a male in that column. Among the 5678 cigar smokers,
there are 4845 males, so the probability we seek is 4845/5678 = 0.85329341. That is,
$P(M | C)$ = 4845/5678 = 0.85329341 = 0.853 (rounded).

**Activity 1:** Now, your turn: the actual population of Boone county is 170,733 (as of 2013). Create the above table with actual population values for the given percentages and find the actual $P(M | C)$.

In [16]:
pop = 170733
m = round(0.51*pop)
f = round(0.49*pop)
ms = round(m * 0.095)
mns = m - ms
fs = round(f * 0.017)
fns = f - fs
cigar2 <- matrix(c(ms, fs, mns, fns), ncol=2)
colnames(cigar2) <- c('smoker','nonsmoker')
rownames(cigar2) <- c('male','female')
cigar2.table <- as.table(cigar2)
cigar2.table/170733
addmargins(cigar2.table/170733)
PMC = ms / (ms+fs)
print(PMC)

            smoker   nonsmoker
male   0.048449919 0.461551077
female 0.008328794 0.481670210

smoker   nonsmoker         Sum
male   0.048449919 0.461551077 0.510000996
female 0.008328794 0.481670210 0.489999004
Sum    0.056778713 0.943221287 1.000000000

[1] 0.8533113


Not surprisingly, it's the same probability as in the first table! Since the proportions did not change, we would not expect this probability change either.

a) Now, using the same table, randomly select an individual, what is the prior probability that the selected person is a female?

b) You later learn that the randomly selected person was smoking a cigar. Use this additional information to find the posterior 
probability that the selected person is a female.

In [17]:
males <- 170733*.51 # number of males
females <- 170733*.49 # number of females
prob_female <- females/(females+males) # prob of females = .49
prob_male <- males/(males+females) # prob of males = .51 

prob_s_female <- (0.008330001) # plug in results using the prop.table function on the cigar dataframe 
prob_s_female = females * 0.017
prob_s_female
prob_s_male <- (0.048450001)
prob_ns_female <- (0.48167)
prob_ns_male <- (0.46155)

numerator <- (prob_female*prob_s_female) # used example from lab on smoking and heart disease and reconstructed the formula
denominator <- ((prob_s_female*prob_s_male)+(prob_ns_female*prob_ns_male)) # use lab example to create inputs for formula 
post <- numerator/denominator 
print(post)

[1] 1422.206

[1] 10.08099


In [18]:
# a) prior probability of a person being female is simply 0.49. That is the percentage of the females in the population.
# If we don't know any extra information, we use that.
print(f/pop)

# b) posterior probability is computed after we learn some extra information; here it is the fact that the person is a smoker.
# We compute P(F|C); the probability that the person is female given that the person is a cigar smoker. 
PFC = fs / (ms+fs)
print(PFC)

[1] 0.489999
[1] 0.1466887


Load the framingham data from the directory '/datasets/framingham'.

In [19]:
framingham_data <- read.csv("../../../datasets/framingham/framingham.csv")
head(framingham_data)

male age education currentSmoker cigsPerDay BPMeds prevalentStroke
1    1  39         4             0          0      0               0
2    0  46         2             0          0      0               0
3    1  48         1             1         20      0               0
4    0  61         3             1         30      0               0
5    0  46         3             1         23      0               0
6    0  43         2             0          0      0               0
  prevalentHyp diabetes totChol sysBP diaBP   BMI heartRate glucose TenYearCHD
1            0        0     195 106.0    70 26.97        80      77          0
2            0        0     250 121.0    81 28.73        95      76          0
3            0        0     245 127.5    80 25.34        75      70          0
4            1        0     225 150.0    95 28.58        65     103          1
5            0        0     285 130.0    84 23.10        85      85          0
6            1        0     228 180.0   110 30.30        77      99          0

**Activity 2:** Create a two-way table from this data set with diabetes condition in the columns and gender in the rows.


In [20]:
dia <- with(framingham_data,table(male,diabetes))
colnames(dia) <- c('nondiabetes','diabetes')
rownames(dia) <- c('female','male')
addmargins(dia)

diabetes
male     nondiabetes diabetes  Sum
  female        2363       57 2420
  male          1768       52 1820
  Sum           4131      109 4240

**Activity 3: **What is the probability for a female to have diabetes? Let d be an event of diabetes and d' be event of nondiabetes. Similarly let g be the event of male and g' be event of female. Find $P(d | g')$ using Bayes formula.

            
                    p(d) * p(g'/d)
     p(d/g') =  -------------------------------------
               [p(d) * p(g'/d)] + [ p(d') * p(g'/d')]
            

In [21]:
# before using the formula, let's apply what we have done in the previous activities; P(d|g') is ratio of diabetes among females
# let's call it PDF 
PDF = 57/2420
print(PDF)


[1] 0.02355372


Let's change the variable names in the formula to make it easier to code:

f is female; nd is nondiabetes

                          p(d) * p(f/d)
     p(d/g') =  -------------------------------------
               [p(d) * p(f/d)] + [ p(nd) * p(f/nd)]
            

In [26]:
#Now use the formula:
addmargins(prop.table(dia))
0.01344340/(0.02570755)

diabetes
male     nondiabetes   diabetes        Sum
  female  0.55731132 0.01344340 0.57075472
  male    0.41698113 0.01226415 0.42924528
  Sum     0.97429245 0.02570755 1.00000000

[1] 0.5229359

In [27]:
PD = 0.02571
PFD = 0.01344 / PD
PND = 0.97429
PFND = 0.55731 / PND

PDF2 = PD*PFD / (PD*PFD + PND*PFND)
print(PDF2)

[1] 0.5227538

[1] 0.5720165

[1] 0.02354796


It is the same result as we obtained before (except some rounding error).